src/device/all_reduce.h

```diff
@@ -362,7 +362,7 @@ namespace {
 #else
   __device__ __attribute__((noinline)) void runTreeSplit(int tid, int nthreads, struct ncclDevWorkColl* work) {
 #endif
-    const int bid = ncclShmem.channelId - work->channelLo;
+    [[maybe_unused]] const int bid = ncclShmem.channelId - work->channelLo;
```

Comment:
It looks like bid is used when ENABLE_NPKIT is defined. Maybe more appropriate fix to wrap the variable with ENABLE_NPKIT .

Reply:
Fixed.